--- gzip-1.9/zforce.in.0001~	2018-01-08 20:52:59.546672777 +0100
+++ gzip-1.9/zforce.in	2018-01-08 20:53:47.080809205 +0100
@@ -56,7 +56,7 @@ for i do
     continue
   fi
 
-  if gzip -lv < "$i" 2>/dev/null | grep '^defl' > /dev/null; then
+  if gzip -l < "$i" 2>/dev/null | grep '^compressed' > /dev/null; then
 
     new="$i.gz"
     mv "$i" "$new" && printf '%s\n' "$i -- replaced with $new" || res=1
