diff -Naur gzip-1.5/zgrep.in gzip-1.5.oden/zgrep.in
--- gzip-1.5/zgrep.in	2012-01-01 09:53:58.000000000 +0100
+++ gzip-1.5.oden/zgrep.in	2012-06-20 16:18:45.000000000 +0200
@@ -171,7 +171,7 @@
 
 exec 3>&1
 res=0
-
+trap break SIGPIPE
 for i
 do
   # Fail if gzip or grep (or sed) fails.
@@ -217,5 +217,8 @@
   test 128 -lt $r && exit $r
   test "$gzip_status" -eq 0 || test "$gzip_status" -eq 2 || r=2
   test $res -lt $r && res=$r
+  # SIGPIPE + 128
+  test "$r" -eq 141 && exit $res 
 done
+trap - SIGPIPE
 exit $res
